code/game/machinery/teleporter.dm

```diff
@@ -5,12 +5,15 @@
 	icon_keyboard = "teleport_key"
 	circuit = /obj/item/circuitboard/teleporter
 	var/obj/item/gps/locked = null
-	var/regime_set = "Teleporter"
+	var/isGate = FALSE // switches behavior between teleporter and gate. Gate = false, teleporter = true
```

Comment:
Please use [snake case](https://en.wikipedia.org/wiki/Snake_case) for your DM variable names